test/e2e_tests/e2e_test.go

```diff
@@ -158,32 +159,68 @@ var _ = Describe("E2E tests", func() {
-			baseNfsPvc := mock.CreateBaseNfsPvc()
-			baseNfsPvc.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{"not good"}
-			Expect(utilst.CreateResource(k8sClient, baseNfsPvc)).Should(BeFalse())
-		})
-	})
 	Context("when PVC with same name already exists", func() {
 		It("should be denied", func() {
 			baseNfsPvc := mock.CreateBaseNfsPvc()
 			pvc := mock.CreateBasePVC(baseNfsPvc.Name)
 			By("creating PVC")
 			Expect(utilst.CreateResource(k8sClient, pvc)).Should(BeTrue())
-			time.Sleep(time.Second * 3)
+			checkPvc := &corev1.PersistentVolumeClaim{}
+			Eventually(func() bool {
+				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: baseNfsPvc.Name, Namespace: baseNfsPvc.Namespace}, checkPvc)
+
+				return err == nil
+			}, TimeoutNfsPvc, NfsPvcCreationInterval).Should(BeTrue())
 			By("creating NFSPVC with same name")
 			Expect(utilst.CreateResource(k8sClient, baseNfsPvc)).Should(BeFalse())
 		})
 	})
-	Context("updating a NFSPVC", func() {
-		It("should be denied", func() {
-			By("creating nfspvc object")
-			baseNfsPvc := mock.CreateBaseNfsPvc()
-			desiredNfsPvc := utilst.CreateNfsPvc(k8sClient, baseNfsPvc)
-			time.Sleep(time.Second * 3)
-			desiredNfsPvc.Spec.Server = "vs-change"
-			Expect(k8sClient.Update(context.Background(), desiredNfsPvc)).ToNot(Succeed(), "failed to update NFSPVC.")
+})
+
+var _ = Describe("Validate CEL object validations", func() {
+	baseNfsPvc := mock.CreateBaseNfsPvc()
+	pvc := &nfspvcv1alpha1.NfsPvc{}
+	baseNfsPvc.Name = "cel-test-pvc"
+	Context("with edit spec", func() {
+		It("creating base pvc", func() {
+
+			Expect(utilst.CreateResource(k8sClient, baseNfsPvc)).Should(BeTrue())
+
+			Eventually(func() bool {
+				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: baseNfsPvc.Name, Namespace: baseNfsPvc.Namespace}, pvc)
+				if err != nil {
+					return false
+				}
+				if pvc.Status.PvcPhase == "Bound" {
+					return true
+				}
+				return false
+			}, TimeoutNfsPvc, NfsPvcCreationInterval).Should(BeTrue())
+		})
+		It("should be denied when editing access mode", func() {
+			pvcCopy := pvc.DeepCopy()
+			pvcCopy.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{"ReadWriteOnce"}
+			err := utilst.UpdateResource(k8sClient, pvcCopy)
+			Expect(err).To(HaveOccurred())
+
+		})
+		It("should be denied when editing capacity", func() {
+			pvcCopy := pvc.DeepCopy()
+			pvcCopy.Spec.Capacity = corev1.ResourceList{corev1.ResourceStorage: resource.MustParse("300Gi")}
+			err := utilst.UpdateResource(k8sClient, pvcCopy)
+			Expect(err).To(HaveOccurred())
+		})
+		It("should be denied when editing path", func() {
+			pvcCopy := pvc.DeepCopy()
+			pvcCopy.Spec.Path = "/update"
+			err := utilst.UpdateResource(k8sClient, pvcCopy)
+			Expect(err).To(HaveOccurred())
+		})
+		It("should be denied when editing server", func() {
+			pvcCopy := pvc.DeepCopy()
+			pvcCopy.Spec.Server = "vs-updated"
+			err := utilst.UpdateResource(k8sClient, pvcCopy)
+			Expect(err).To(HaveOccurred())
```

Comment:
I think it would be better to do all these tests in a single `It` block just with different `By` blocks. `It` blocks can run in parallel so you shouldn't assume data and variable locality between `It` blocks.